src/test/java/org/broadinstitute/hellbender/tools/examples/ExampleVariantWalkerIntegrationTest.java

```diff
@@ -13,17 +14,30 @@
     private static final String TEST_OUTPUT_DIRECTORY = publicTestDir + "org/broadinstitute/hellbender/tools/examples/";
 
     @Test
-    public void testExampleVariantWalker() throws IOException {
-        IntegrationTestSpec testSpec = new IntegrationTestSpec(
+    public void testExampleVariantWalkerNoSeqDict() throws IOException {
+        final IntegrationTestSpec testSpec = new IntegrationTestSpec(
                 " -L 1:100-200" +
                 " -R " + hg19MiniReference +
                 " -I " + TEST_DATA_DIRECTORY + "reads_data_source_test1.bam" +
                 " -V " + TEST_DATA_DIRECTORY + "example_variants.vcf" +
                 " -auxiliaryVariants " + TEST_DATA_DIRECTORY + "feature_data_source_test.vcf" +
                 " -O %s",
+                1,
+                UserException.MalformedGenomeLoc.class);
```

Comment:
the seq dictionary that gets used is the one from the index and it has contig lengths that are 0 (undefined).  The bam one is not used. The exception is caused by trying to do -L 1:100-200 on a contig of length 0.

I think this is a bug, but for a different report. I think the desired behavior may be either of:
a) prefer the bam dictionary because it has defined contig lengths
b) create an uber-dictionary from all compatible dictionaries that are available
c) make it work with 0-length contigs by accepting any -L queries to them. 
I'll leave it up to you to pick the right solution and enter a bug report for it.